examples/api/test/material/refresh_indicator/refresh_indicator.2_test.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/refresh_indicator/refresh_indicator.2.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Pulling from scroll view triggers a refresh indicator which shows a CircularProgressIndicator', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.RefreshIndicatorExampleApp(),
+    );
+
+    // Pull the first item
+    await tester.fling(find.text('Pull down here').first, const Offset(0.0, 300.0), 1000.0);
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+    await tester.pump(const Duration(seconds: 1));
+
+    expect(find.byType(RefreshProgressIndicator), findsNothing);
+    expect(find.bySemanticsLabel('Circular progress indicator'), findsOneWidget);
+
+    await tester.pumpAndSettle(); // Advance pending time
```

Comment:
Can this be done using pumps with a duration? https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-pumpandsettle

Reply:
Nit: Comment should also end in a period.

Reply:
I couldn't get it to work using durations, I honestly don't know why.

Reply:
It's fine if not. I was just asking if you explored those options. 

Maybe something like a first pump with the animation duration, and an additional pump with a small time period to advance past the animation could work?